core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2852,18 +2853,23 @@ struct ViewValueFunctor<ExecSpace, ValueType, false /* is_scalar */> {
   ViewValueFunctor& operator=(const ViewValueFunctor&) = default;
 
   ViewValueFunctor(ExecSpace const& arg_space, ValueType* const arg_ptr,
-                   size_t const arg_n)
-      : space(arg_space), ptr(arg_ptr), n(arg_n), destroy(false) {}
+                   size_t const arg_n, std::string arg_name)
+      : space(arg_space),
+        ptr(arg_ptr),
+        n(arg_n),
+        destroy(false),
+        name(std::move(arg_name)) {}
 
   void execute(bool arg) {
     destroy = arg;
     if (!space.in_parallel()) {
+      std::string functor_name = (destroy ? "Kokkos::View::destruction"
+                                          : "Kokkos::View::initialization");
+      functor_name += " [" + name + "]";
```

Comment:
Why isn't this just integrated in to the string literals above? Also, besides the fact that it seems like a waste of literal entries in the binary to put it here, this potentially does an extra _two_ allocations, which seems bad.